plugin/sampling/strategystore/static/strategy_store_test.go

```diff
@@ -333,7 +333,6 @@ func TestDeepCopy(t *testing.T) {
 		},
 	}
 	cp := deepCopy(s)
-	assert.False(t, cp == s)
```

Comment:
this seems important - it was ensuring the two structs are not the same, even if equal in content

Reply:
can we fix this?

Reply:
If I try to comply with testifylint it will be 
assert.NotEqual(t, cp, s) but this fails .
Maybe I am applying the wrong correction .
Any better idea?

Reply:
Is there something like Same() check?

Reply:
Yes Same and NotSame